### ISSUE TITLE
Model.fit doesn't train properly likely due to GL_INVALID_OPERATION error

### ISSUE THREAD

**Issue by Chrisvin** [tensorflow/tfjs#1381]
I followed the steps from "https://codelabs.developers.google.com/codelabs/tfjs-training-classfication/index.html?index=..%2F..index#0" to make CNN that can be used for digit recognition.

The model used to work, but it doesn't work now. Some sort of problem occurs during the model's training, the loss and accuracy values (shown using tensorflow/tfjs-vis) drop drastically and "[.WebGL-0000016FAEA3AAA0] GL_INVALID_OPERATION: Object cannot be used because it has not been generated." is thrown multiple times in the browser console.
![image](https://user-images.githubusercontent.com/20887005/54338551-38788680-4658-11e9-80fb-fd4aa524a07c.png)

Once the training completes and evaluation of the model (using tjfs-vis per class accuracy and confusion matrix) begins, the screen completely blacks out for a split second and the expected output is not generated.
![image](https://user-images.githubusercontent.com/20887005/54338834-061b5900-4659-11e9-841b-0b22bb65e369.png)

I have not changed the code since I last checked this a few weeks ago(when it worked perfectly).
And the same issue arises in another project I am working on as well.
Any help or advice as to what's going wrong would be much appreciated.

**Comment by caisq** [comment]
@Chrisvin Thanks for reporting this issue.

What device are you using?What is your operating system and chrome version?

**Comment by Chrisvin** [comment]
Device - Dell Latitude 7490 (I have also checked on a Lenovo Yoga 720 which also had the same issues)
OS - Windows 10
Chrome - Version 72.0.3626.121 (Official Build) (64-bit)

**Comment by caisq** [comment]
@Chrisvin Just to cover the ground, have you tried restarting Chrome and/or rebooting system and see whether it resolves the problem?

**Comment by Chrisvin** [comment]
Yes, I've tried restarting chrome, clearing it's cache and rebooting the system.
The reason I tried on a different laptop was to try to figure out if it was a problem related to my system.

**Comment by Chrisvin** [comment]
@caisq Also, I have used the script tag as follows to use TensorFlow.js and tfjs-vis.

```
<script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@1.0.0/dist/tf.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs-vis@1.0.2/dist/tfjs-vis.umd.min.js"></script>
```
I have also tried using older versions as follows but it also results in the same issues, 
```
<script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@0.14.2/dist/tf.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs-vis@0.4.2/dist/tfjs-vis.umd.min.js"></script>
```

**Comment by Chrisvin** [comment]
@caisq Just to ensure that I'm not doing something dumb with my code, I've checked some of the [TensorFlow.js examples](https://github.com/tensorflow/tfjs-examples) and some of the examples don't work properly.

For example, the [MNIST example](https://storage.googleapis.com/tfjs-examples/mnist/dist/index.html) doesn't train properly and exhibits weird/unexpected behavior ('random' drops in accuracy and loss values, split second screen blackening and multiple graph creation) as shown below,
![image](https://user-images.githubusercontent.com/20887005/54383890-1400c680-46b9-11e9-9feb-93b0b72a3701.png)
Similar issues can be seen in [MNIST CNN Transfer Learning example](https://storage.googleapis.com/tfjs-examples/mnist-transfer-cnn/dist/index.html).

This leads me to believe that the issue is on the side of tensorflow.js,
Any updates related to this would be appreciated.

**Comment by caisq** [comment]
cc @dsmilkov @nsthorat for thoughts on the lower-level WebGL issue.

**Comment by bileschi** [comment]
@dsmilkov : can you triage and assign this to the right person / priority?  Thanks.

**Comment by nsthorat** [comment]
Thanks for reporting this issue -- we're working on getting our windows machine up and running and will get back to you ASAP.

**Comment by caisq** [comment]
@Chrisvin Just a thought: what graphic(s) card do you have on your machine? Is it possible that there are two and switching to another one in your Windows settings might help?

@nsthorat @dsmilkov

**Comment by Chrisvin** [comment]
@caisq 
The Dell Latitude 7490 has a Intel(R) UHD Graphics 620.
I can't check the Lenovo Yoga 720 atm, but I believe that it has a Intel HD Graphics 620.
So, no, there isn't two graphic cards. Besides, all of these examples and models used to work perfectly well, so I'm assuming that they were fully capable of handling the tasks, or am I misunderstanding something?

**Comment by dsmilkov** [comment]
Hi @Chrisvin , we are waiting for a windows machine to arrive so we can reproduce this problem.

In the meantime, since you mentioned that *this used to work*, if you can test and find the latest version of tf.js that makes your demo work (0.15.3, 0.15.2, 0.15.1, 0.15.0, 0.14.2,...?), that would be very helpful to us. Thank you!

**Comment by Chrisvin** [comment]
@caisq @nsthorat @nsthorat
The TensorFlow.js examples and my projects have started working properly again. So, I guess this issue can be closed now.
Thanks for the quick replies and support guys.

**Comment by dsmilkov** [comment]
Thanks Chris, I'll close the issue - feel free to reopen if you see the problem again. If you find out what changed this time vs last time, please share it with us so we can improve our library and sure this doesn't happen again.
